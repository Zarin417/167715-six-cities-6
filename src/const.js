export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const CardName = {
  CITIES: `cities`,
  FAVORITES: `favorites`,
  NEAR_PLACES: `near-places`,
};

export const CardsListName = {
  NEAR_PLACES_LIST: `near-places__list`,
  CITIES_PLACES_LIST: `cities__places-list tabs__content`
};

export const ButtonName = {
  PROPERTY: `property`,
  PLACE_CARD: `place-card`,
  FAVORITE: `favorite`,
  NEAR_PLACE: `near-place`
};

export const SortType = {
  POPULAR: `popular`,
  PRICE_LOW_TO_HIGH: `price: low to high`,
  PRICE_HIGH_TO_LOW: `price: high to low`,
  TOP_RATED: `top rated first`
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  CITY: `/city/:city`,
  OFFER: `/offer/:id`,
  ERROR: `/error`
};

export const APIRoute = {
  HOTELS: `/hotels`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};
