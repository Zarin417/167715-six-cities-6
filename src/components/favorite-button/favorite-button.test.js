import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import FavoriteButton from "./favorite-button";
import {ButtonName} from "../../const";

const mockStore = configureStore({});

describe(`Test 'FavoriteButton'`, () => {
  it(`FavoriteButton in the OfferScreen component should render correctly`, () => {
    const store = mockStore({
      PLACE_INFO: {placeInfo: {}},
      USER: {isLoggedIn: true}
    });

    const {container} = render(
        <Provider store={store}>
          <FavoriteButton
            isFavorite={true}
            buttonName={ButtonName.PROPERTY}
            placeId={99}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`FavoriteButton in the Offer component should render correctly`, () => {
    const store = mockStore({
      PLACE_INFO: {placeInfo: {}},
      USER: {isLoggedIn: true}
    });

    const {container} = render(
        <Provider store={store}>
          <FavoriteButton
            isFavorite={false}
            buttonName={ButtonName.PLACE_CARD}
            placeId={42}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
