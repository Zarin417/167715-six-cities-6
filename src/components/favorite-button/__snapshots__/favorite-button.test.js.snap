// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'FavoriteButton' FavoriteButton in the Offer component should render correctly 1`] = `
<div>
  <button
    class="place-card__bookmark-button
       button"
    type="button"
  >
    <svg
      class="place-card__bookmark-icon"
      height="19"
      width="18"
    >
      <use
        xlink:href="#icon-bookmark"
      />
    </svg>
    <span
      class="visually-hidden"
    >
      To bookmarks
    </span>
  </button>
</div>
`;

exports[`Test 'FavoriteButton' FavoriteButton in the OfferScreen component should render correctly 1`] = `
<div>
  <button
    class="property__bookmark-button
      property__bookmark-button--active button"
    type="button"
  >
    <svg
      class="property__bookmark-icon"
      height="33"
      width="31"
    >
      <use
        xlink:href="#icon-bookmark"
      />
    </svg>
    <span
      class="visually-hidden"
    >
      To bookmarks
    </span>
  </button>
</div>
`;
